GenderPayGap.WebUI/Controllers/Admin/AdminOrganisationSectorController.cs

```diff
@@ -28,6 +30,71 @@ public class AdminOrganisationSectorController : Controller
             this.dataRepository = dataRepository;
             this.auditLogger = auditLogger;
         }
+        
+          [HttpGet("organisation/{id}/sector")]
+        public IActionResult ViewSectorHistory(long id)
+        {
+            var viewModel = new AdminSectorHistoryViewModel();
+            
+            var organisation = dataRepository.Get<Organisation>(id);
+            var sectorHistory = dataRepository.GetAll<AuditLog>()
+                .Where(al => al.Action == AuditedAction.AdminChangedOrganisationSector)
+                .Where(al => al.Organisation.OrganisationId == id)
+                .OrderByDescending(al => al.CreatedDate)
+                .ToList();
+
+            viewModel.Organisation = organisation;
+            viewModel.SectorHistory = sectorHistory;
+
+            return View("ViewOrganisationSector", viewModel);
+        }
+        
+        [HttpGet("organisation/{id}/sector/change")]
+        public IActionResult ChangeSectorGet(long id)
+        {
+            var viewModel = new AdminChangeSectorViewModel();
+
+            UpdateAdminChangeSectorViewModelFromOrganisation(viewModel, id);
+            viewModel.NewSector = viewModel.Organisation.SectorType == SectorTypes.Private ? NewSectorTypes.Private : NewSectorTypes.Public;
+
+            return View("ChangeSector", viewModel);
+        }
+        
+        [HttpPost("organisation/{id}/sector/change")]
+        [PreventDuplicatePost]
+        [ValidateAntiForgeryToken]
+        public IActionResult ChangeSectorPost(long id, AdminChangeSectorViewModel viewModel)
+        {
+            switch (viewModel.Action)
+            {
+                case ChangeOrganisationSectorViewModelActions.OfferNewSectorAndReason:
+                    UpdateAdminChangeSectorViewModelFromOrganisation(viewModel, id);
+                    ValidateAdminChangeSectorViewModel(viewModel);
+                    
+                    // Check if new sector is same as original organisation sector
+                    var newSector = viewModel.NewSector == NewSectorTypes.Private ? SectorTypes.Private : SectorTypes.Public;
+                    if (newSector == viewModel.Organisation.SectorType)
+                    {
+                        viewModel.AddErrorFor(
+                            m => m.NewSector,
+                            "The organisation is already assigned to this sector."
+                        );
+                    }
+                    
+                    if (viewModel.HasAnyErrors())
+                    {
+                        return View("ChangeSector", viewModel);
+                    }
+            
+                    return View("ConfirmSectorChange", viewModel);
+            
+                case ChangeOrganisationSectorViewModelActions.ConfirmSectorChange:
+                    ChangeSector(viewModel, id);
+                    return RedirectToAction("ViewSectorHistory", "AdminOrganisationSector", new {id});

```

Comment:
Minor:
I think this code might be a bit easier to read if we move the body of each `case` into a separate method - that way, it's a bit clearer that each case is responding to a different action, rather than it looking like they might both be from the same page